### ISSUE TITLE
Training loss NaN when using multi-head graph attention model

### ISSUE THREAD

**Issue by zhangyu2ustc** [pyg-team/pytorch_geometric#351]
## 🐛 Bug

I used the GAT model for classification. When I set the head to 1, the training process is fine and quickly converge. But if I increase the head to 2 or bigger, the training loss became from the first epoch
## To Reproduce

1. using GAT with head=1, the performance converged to 90%
1. increase head to 2 with either concate or not, the training loss became nan from epoch 1
1.

## Environment

* OS: ubuntu
* Python version:3.6
* PyTorch version: 1.1.0
* CUDA/cuDNN version:8.0
* GCC version:
* Any other relevant information:

**Comment by rusty1s** [comment]
Hi, do you have a concrete example to reproduce this? The `gat.py` example uses multi-head attention, so I am wondering what might cause this issue.
